runtime/compiler/control/J9Options.cpp

```diff
@@ -2028,30 +2039,40 @@ J9::Options::fePreProcess(void * base)
    if (!TR::Compiler->target.cpu.isZ())
       self()->setOption(TR_DisableAOTBytesCompression);
 
-   // Check for option -XX:JITaaSClient and/or -XX:JITaaSServer
-   static bool JITaaSAlreadyParsed = false;
-   if (!JITaaSAlreadyParsed) // avoid processing twice for AOT and JIT and produce duplicate messages
+#if defined(JITSERVER_SUPPORT)
+   // Check for option -XX:+UseJITServer and/or -XX:+StartAsJITServer
+   // -XX:-UseJITServer or -XX:-StartAsJITServer disables JITServer
+   static bool JITServerAlreadyParsed = false;
+   if (!JITServerAlreadyParsed) // avoid processing twice for AOT and JIT and produce duplicate messages
       {
-      JITaaSAlreadyParsed = true;
-      char *xxJITaaSClientOption = "-XX:JITaaSClient";
-      char *xxJITaaSServerOption = "-XX:JITaaSServer";
-      int32_t xxJITaaSClientArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxJITaaSClientOption, 0);
-      int32_t xxJITaaSServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxJITaaSServerOption, 0);
+      JITServerAlreadyParsed = true;
+
+      char *xxUseJITServerOption = "-XX:+UseJITServer";
+      char *xxDisableUseJITServerOption = "-XX:-UseJITServer";
+      char *xxStartAsJITServerOption = "-XX:+StartAsJITServer";
+      char *xxDisableStartAsJITServerOption = "-XX:-StartAsJITServer";
+
+      int32_t xxUseJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxUseJITServerOption, 0);
+      int32_t xxDisableUseJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxDisableUseJITServerOption, 0);
+      int32_t xxStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxStartAsJITServerOption, 0);
+      int32_t xxDisableStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxDisableStartAsJITServerOption, 0);
+
       // Check if option is at all specified
-      if (xxJITaaSClientArgIndex >= 0 || xxJITaaSServerArgIndex >= 0)
+      if (((xxUseJITServerArgIndex > xxDisableUseJITServerArgIndex ) && (xxUseJITServerArgIndex >= 0)) ||
```

Comment:
There is an extra space before closing parenthesis.
Do you need the check for  `(xxUseJITServerArgIndex >= 0)`? I am guessing that if no option is present then `xxUseJITServerArgIndex` has the same value as `xxDisableUseJITServerArgIndex` and the part after `&&` does not get evaluated.

Reply:
You're right. No need to compare `(xxUseJITServerArgIndex >= 0)`